cpp/src/text/edit_distance.cu

```diff
@@ -137,154 +129,157 @@ struct edit_distance_matrix_levenshtein_algorithm {
+  thrust::uninitialized_fill(rmm::exec_policy_nosync(stream), offsets.begin(), offsets.end(), 0);
+  thrust::for_each_n(rmm::exec_policy_nosync(stream),
+                     thrust::counting_iterator<cudf::size_type>(0),
+                     output_size,
+                     calculate_matrix_compute_buffer_fn{*d_strings, offsets.data()});
 
   // get the total size for the compute buffer
-  int64_t compute_size =
-    thrust::reduce(rmm::exec_policy(stream), offsets.begin(), offsets.end(), int64_t{0});
-  // convert sizes to offsets in-place
-  thrust::exclusive_scan(rmm::exec_policy(stream), offsets.begin(), offsets.end(), offsets.begin());
+  // and convert sizes to offsets in-place
+  auto const compute_size =
+    cudf::detail::sizes_to_offsets(offsets.begin(), offsets.end(), offsets.begin(), 0, stream);
+
   // create the compute buffer
   rmm::device_uvector<cudf::size_type> compute_buffer(compute_size, stream);
   auto d_buffer = compute_buffer.data();
 
   // compute the edit distance into the output column
-  auto results = cudf::make_fixed_width_column(cudf::data_type{cudf::type_to_id<cudf::size_type>()},
-                                               strings_count * strings_count,
-                                               rmm::device_buffer{0, stream, mr},
-                                               0,
-                                               stream,
-                                               mr);
+  auto results = cudf::make_fixed_width_column(
+    output_type, output_size, rmm::device_buffer{0, stream, mr}, 0, stream, mr);
   auto d_results = results->mutable_view().data<cudf::size_type>();
   thrust::for_each_n(
-    rmm::exec_policy(stream),
-    thrust::make_counting_iterator<cudf::size_type>(0),
-    strings_count * strings_count,
-    edit_distance_matrix_levenshtein_algorithm{d_strings, d_buffer, d_offsets, d_results});
+    rmm::exec_policy_nosync(stream),
+    thrust::counting_iterator<cudf::size_type>(0),
+    output_size,
+    edit_distance_matrix_levenshtein_algorithm{*d_strings, d_buffer, offsets.data(), d_results});
 
   // build a lists column of the results
   auto offsets_column =
-    cudf::make_fixed_width_column(cudf::data_type{cudf::type_to_id<cudf::size_type>()},
-                                  strings_count + 1,
-                                  rmm::device_buffer{0, stream, mr},
-                                  0,
-                                  stream,
-                                  mr);
-  thrust::transform_exclusive_scan(
-    rmm::exec_policy(stream),
-    thrust::counting_iterator<cudf::size_type>(0),
-    thrust::counting_iterator<cudf::size_type>(strings_count + 1),
-    offsets_column->mutable_view().data<cudf::size_type>(),
-    [strings_count] __device__(auto idx) { return strings_count; },
-    cudf::size_type{0},
-    cuda::std::plus<cudf::size_type>());
-  return cudf::make_lists_column(strings_count,
+    cudf::detail::sequence(input.size() + 1,
```

Comment:
Overflow?

Reply:
`input.size()` must be less than `max<size_type>` so `input.size()+1` will not overflow.